calico_node/startup/startup.go

```diff
@@ -140,36 +148,14 @@ func determineNodeName() string {
 	return nodeName
 }
 
-// createClient() creates the Calico API client, and unless requested otherwise,
-// checks that the client is accessible.
-func createClient() (*api.CalicoAPIConfig, *client.Client) {
-	log.Info("Loading client config")
-	cfg, err := client.LoadClientConfig("")
-	if err != nil {
-		fatal("Error loading datastore config: %s", err)
-		terminate()
-	}
-
-	log.Info("Creating client")
-	client, err := client.New(*cfg)
-	if err != nil {
-		fatal("Error accessing the Calico datastore: %s", err)
-		terminate()
-	}
-
-	// An explicit value of false is required to skip waiting for the
-	// datastore.
-	if os.Getenv("WAIT_FOR_DATASTORE") != "true" {
-		message("Skipping datastore connection test")
-		return cfg, client
-	}
-
+// waitForConnection() waits for the datastore to become accessible.
+func waitForConnection(c *client.Client) {
 	message("Checking datastore connection")
 	for {
 		// Query some arbitrary configuration to see if the connection
 		// is working.  Getting a specific node is a good option, even
 		// if the node does not exist.
-		_, err = client.Nodes().Get(api.NodeMetadata{Name: "foo"})
+		_, err := c.Nodes().Get(api.NodeMetadata{Name: "foo"})
```

Comment:
We could - it's fewer queries to do this I think and EnsureInitialized will be called as part of the node resource management so yeah we could, but it's no biggy either way.